src/promise/pTo.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * Wraps async calls in try catch blocks
+ * to simplify syntax.
+ *
+ * source: https://github.com/scopsy/await-to-js/blob/master/src/await-to-js.ts
+ *
+ * @param { Promise } promise
+ * @param { Object= } errorExt - Additional Information you can pass to the err object
+ * @return { Promise } Tuple with error or response
+ */
+ export async function pTo<T, U = Error>(
```

Comment:
I'll check the NCBackend3 PR with usage examples to give some feedback on the usefulness of this function.

So far I have few comments/concerns:

1. Function name. `pTo` doesn't tell me much about this function. Actually, the conventional `p` prefix tells me that it's something about a Promise. I'm open for suggestions on how to name it better. Something like `pTuple` or `pAwait` or `pResult` or `awaitToJs` (without the `p*` convention).

2. I'm curious to see cases where `errorExt` is useful.

3. It looks like (I may be wrong) wrapping a Promise in this function will NOT keep the "async stacktraces". (see https://v8.dev/blog/fast-async#improved-developer-experience). E.g we have the convention to do `return await` instead of just `return somePromise` to keep the async stacktraces (which proven to be very useful in e.g Sentry or logs). So, `return await` here will maybe fix it. Or, if that's not enough, `try { await promise } catch (e) { ... }` would fix that. Can be checked/proven by a unit test.

Reply:
1. Probably the best name would be to just call it "to" as the original library, as when you wrap an async function it read 'await to(Fx)' very readable. The p was just to follow the convention that we seemed to have on promises. Whatever name I'm fine with it, I just need the functionality :) 

2. Whenever you want to extend the error object with custom values. Out of the test cases of the original library

```
const [err] = await to<
      string,
      { error: string; extraKey: number }
    >(promise, {
      extraKey: 1
    });
```
    
I have not used it, but I'm sure that it must have some use cases  

3. I am not 100% sure about this. I would say that "return await" is not needed unless you have a try-catch block, otherwise "return Promise" and "return await Promise" are the same (there is even an eslint rule to avoid this) as in both cases you are passing the error to the caller unless you do something like
```
try {
return await SomePromise()
}
catch(e) {
  // here you will handle the error inside the function and not pass it to the caller
}
```
As in this case, you would wait for the result/error 

You would use this function only in cases that you want to handle the error, otherwise just using const bar = await Foo should be enough to let it throw the error.
If you wanted to throw the original error you could to something like 
`const [error, response] = await SomePromise()
  if(error) {
     throw error
 }
`

Reply:
> I am not 100% sure about this. I would say that "return await" is not needed unless you have a try-catch block, otherwise "return Promise" and "return await Promise" are the same (there is even an eslint rule to avoid this)

Yes, I know about that eslint-rule, but I consider it outdated since async-stacktraces landed in v8. Please check my link: https://v8.dev/blog/fast-async#improved-developer-experience

The implications of these async-stacktraces are quite important, in my opinion. E.g I can find in Sentry the places where async-stacktraces don't work and the errors there look like "out of nowhere". And there are places where stacktraces are preserved and it looks very nice and is useful.

It is possible to create a unit test where this will be illustrated. I'm confident that it'll show the difference in stack trace of `return Promise` vs `return await Promise`.

Reply:
> 1. Probably the best name would be to just call it "to"

I try to avoid names that are also javascript keywords. Like, `default`.

But wait... Actually, `to` is not a keyword in JS (mixed it with Kotlin, probably), so, not an issue.

On the other hand, I don't find `to` name very descriptive of what it does.

Of the suggestions that I listed yesterday I probably like `pTuple` the most, because... it returns a Tuple!:)
So, at least the reader would expect a Tuple somewhere.

What do you think? Should we settle on `pTuple`?

Reply:
>  Please check my link: https://v8.dev/blog/fast-async#improved-developer-experience

I did a quick check, and it seems you are right :) ...So weird that everyone discourages this practice (many negative comments in StackOverflow)
![image](https://user-images.githubusercontent.com/48744933/112191003-50d6f080-8be4-11eb-84e8-5baba2b207eb.png)

![image](https://user-images.githubusercontent.com/48744933/112191047-5d5b4900-8be4-11eb-9400-e9a6bcd0ae4a.png)

I also checked if the "to" function has any effect on the stack, and it seems that it works just fine, keeping the stack trace as it should be.
![image](https://user-images.githubusercontent.com/48744933/112191959-4832ea00-8be5-11eb-8c84-03bb36556420.png)

> What do you think? Should we settle on pTuple?

Sure pTuple works for me, I will fix the PR with the new name

Reply:
Great!

Yeah, probably `pTo` keeps the stack-trace because it has `.then()` which is kind of the same as "awaiting it". But great that you checked it, so we can keep it as-is then.